panic: runtime error: index out of range [1] with length 1

#### What happened:

The operator immediately crashed after installation with an index out of range error.

```
$ k get pod -n nextcloud 
NAME                                 READY   STATUS      RESTARTS        AGE
nextcloud-metrics-6fc67549b9-nkj54   1/1     Running     3 (5h50m ago)   41h
redis-master-0                       1/1     Running     4 (5h50m ago)   2d22h
mariadb-0                            1/1     Running     2 (5h50m ago)   21h
nextcloud-5c65f8ffd7-8rqfm           2/2     Running     7 (5h50m ago)   41h
nextcloud-cron-27579690-gf6qn        0/1     Completed   0               2m38s

$ k get pod -n security-profiles-operator 
NAME                                          READY   STATUS             RESTARTS     AGE
security-profiles-operator-7bcfcc4589-dbcvj   0/1     CrashLoopBackOff   7 (5m ago)   18m

$ k logs -n security-profiles-operator security-profiles-operator-7bcfcc4589-dbcvj 
I0609 13:25:29.124673       1 logr.go:261]  "msg"="Set logging verbosity to 0"  
I0609 13:25:29.124742       1 logr.go:261]  "msg"="Profiling support enabled: false"  
I0609 13:25:29.124794       1 logr.go:261] setup "msg"="starting component: security-profiles-operator"  "buildDate"="1980-01-01T00:00:00Z" "buildTags"="netgo,osusergo,seccomp,apparmor" "cgoldFlags"="-lseccomp -lelf -lz -lbpf" "compiler"="gc" "dependencies"="github.com/PuerkitoBio/purell v1.1.1 ,github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578 ,github.com/ReneKroon/ttlcache/v2 v2.11.0 ,github.com/acobaugh/osrelease v0.0.0-20181218015638-a93a0a55a249 ,github.com/aquasecurity/libbpfgo v0.2.5-libbpf-0.7.0 ,github.com/beorn7/perks v1.0.1 ,github.com/blang/semver v3.5.1+incompatible ,github.com/cert-manager/cert-manager v1.8.0 ,github.com/cespare/xxhash/v2 v2.1.2 ,github.com/containers/common v0.48.1-0.20220510094751-400832f41771 ,github.com/cpuguy83/go-md2man/v2 v2.0.1 ,github.com/crossplane/crossplane-runtime v0.16.0 ,github.com/davecgh/go-spew v1.1.1 ,github.com/emicklei/go-restful v2.9.5+incompatible ,github.com/evanphx/json-patch v4.12.0+incompatible ,github.com/fsnotify/fsnotify v1.5.1 ,github.com/go-logr/logr v1.2.3 ,github.com/go-openapi/jsonpointer v0.19.5 ,github.com/go-openapi/jsonreference v0.19.5 ,github.com/go-openapi/swag v0.19.14 ,github.com/gogo/protobuf v1.3.2 ,github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da ,github.com/golang/protobuf v1.5.2 ,github.com/google/gnostic v0.5.7-v3refs ,github.com/google/go-cmp v0.5.6 ,github.com/google/gofuzz v1.2.0 ,github.com/google/uuid v1.3.0 ,github.com/imdario/mergo v0.3.12 ,github.com/josharian/intern v1.0.0 ,github.com/json-iterator/go v1.1.12 ,github.com/mailru/easyjson v0.7.6 ,github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369 ,github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd ,github.com/modern-go/reflect2 v1.0.2 ,github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 ,github.com/nxadm/tail v1.4.8 ,github.com/opencontainers/runtime-spec v1.0.3-0.20210326190908-1c3f411f0417 ,github.com/openshift/api v0.0.0-20220209124712-b632c5fc10c0 ,github.com/pjbgf/go-apparmor v0.0.7 ,github.com/pkg/errors v0.9.1 ,github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.57.0 ,github.com/prometheus/client_golang v1.12.2 ,github.com/prometheus/client_model v0.2.0 ,github.com/prometheus/common v0.32.1 ,github.com/prometheus/procfs v0.7.3 ,github.com/russross/blackfriday/v2 v2.1.0 ,github.com/seccomp/libseccomp-golang v0.9.2-0.20210429002308-3879420cc921 ,github.com/sirupsen/logrus v1.8.1 ,github.com/spf13/afero v1.8.0 ,github.com/spf13/pflag v1.0.5 ,github.com/urfave/cli/v2 v2.8.1 ,github.com/xrash/smetrics v0.0.0-20201216005158-039620a65673 ,golang.org/x/net v0.0.0-20220225172249-27dd8689420f ,golang.org/x/oauth2 v0.0.0-20211104180415-d3ed0bb246c8 ,golang.org/x/sync v0.0.0-20210220032951-036812b2e83c ,golang.org/x/sys v0.0.0-20220422013727-9388b58f7150 ,golang.org/x/term v0.0.0-20210927222741-03fcf44c2211 ,golang.org/x/text v0.3.7 ,golang.org/x/time v0.0.0-20220210224613-90d013bbcef8 ,gomodules.xyz/jsonpatch/v2 v2.2.0 ,google.golang.org/genproto v0.0.0-20220304144024-325a89244dc8 ,google.golang.org/grpc v1.47.0 ,google.golang.org/protobuf v1.28.0 ,gopkg.in/inf.v0 v0.9.1 ,gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7 ,gopkg.in/yaml.v2 v2.4.0 ,gopkg.in/yaml.v3 v3.0.1 ,k8s.io/api v0.24.1 ,k8s.io/apiextensions-apiserver v0.24.0 ,k8s.io/apimachinery v0.24.1 ,k8s.io/client-go v0.24.1 ,k8s.io/component-base v0.24.0 ,k8s.io/klog/v2 v2.60.1 ,k8s.io/kube-openapi v0.0.0-20220328201542-3ee0da9b0b42 ,k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9 ,sigs.k8s.io/controller-runtime v0.12.1 ,sigs.k8s.io/gateway-api v0.4.1 ,sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2 ,sigs.k8s.io/release-utils v0.7.0 ,sigs.k8s.io/structured-merge-diff/v4 v4.2.1 ,sigs.k8s.io/yaml v1.3.0 " "gitCommit"="unknown" "gitCommitDate"="unknown" "gitTreeState"="clean" "goVersion"="go1.18.2" "ldFlags"="-s -w -linkmode external -extldflags \"-static\" -X sigs.k8s.io/security-profiles-operator/internal/pkg/version.buildDate=1980-01-01T00:00:00Z -X sigs.k8s.io/security-profiles-operator/internal/pkg/version.version=0.4.3" "libbpf"="v0.7" "libseccomp"="2.5.3" "platform"="linux/amd64" "version"="0.4.3"
I0609 13:25:29.978574       1 logr.go:261] controller-runtime/metrics "msg"="Metrics server is starting to listen"  "addr"=":8080"
I0609 13:25:29.985416       1 logr.go:261] setup "msg"="starting manager"  
I0609 13:25:29.985659       1 internal.go:362]  "msg"="Starting server" "addr"={"IP":"::","Port":8080,"Zone":""} "kind"="metrics" "path"="/metrics" 
I0609 13:25:30.085976       1 leaderelection.go:248] attempting to acquire leader lease security-profiles-operator/security-profiles-operator-lock...
I0609 13:25:48.000646       1 leaderelection.go:258] successfully acquired lease security-profiles-operator/security-profiles-operator-lock
I0609 13:25:48.001494       1 controller.go:185]  "msg"="Starting EventSource" "controller"="pods" "controllerGroup"="" "controllerKind"="Pod" "source"="kind source: *v1alpha1.SecurityProfileNodeStatus"
I0609 13:25:48.001516       1 controller.go:185]  "msg"="Starting EventSource" "controller"="pods" "controllerGroup"="" "controllerKind"="Pod" "source"="kind source: *v1alpha1.SecurityProfilesOperatorDaemon"
I0609 13:25:48.001565       1 controller.go:193]  "msg"="Starting Controller" "controller"="pods" "controllerGroup"="" "controllerKind"="Pod" 
I0609 13:25:48.001597       1 controller.go:185]  "msg"="Starting EventSource" "controller"="pods" "controllerGroup"="" "controllerKind"="Pod" "source"="kind source: *v1.DaemonSet"
I0609 13:25:48.001640       1 controller.go:193]  "msg"="Starting Controller" "controller"="pods" "controllerGroup"="" "controllerKind"="Pod" 
I0609 13:25:48.001800       1 controller.go:185]  "msg"="Starting EventSource" "controller"="pods" "controllerGroup"="" "controllerKind"="Pod" "source"="kind source: *v1.Pod"
I0609 13:25:48.001839       1 controller.go:193]  "msg"="Starting Controller" "controller"="pods" "controllerGroup"="" "controllerKind"="Pod" 
I0609 13:25:48.103168       1 controller.go:227]  "msg"="Starting workers" "controller"="pods" "controllerGroup"="" "controllerKind"="Pod" "worker count"=1
I0609 13:25:48.103185       1 controller.go:227]  "msg"="Starting workers" "controller"="pods" "controllerGroup"="" "controllerKind"="Pod" "worker count"=1
I0609 13:25:48.103177       1 controller.go:227]  "msg"="Starting workers" "controller"="pods" "controllerGroup"="" "controllerKind"="Pod" "worker count"=1
I0609 13:25:48.103686       1 controller.go:117]  "msg"="Observed a panic in reconciler: runtime error: index out of range [1] with length 1" "controller"="pods" "controllerGroup"="" "controllerKind"="Pod" "name"="mariadb-0" "namespace"="nextcloud" "pod"={"name":"mariadb-0","namespace":"nextcloud"} "reconcileID"="c2d975c7-2fe5-4771-8e90-1e05fe43d2b3" 
panic: runtime error: index out of range [1] with length 1 [recovered]
        panic: runtime error: index out of range [1] with length 1

goroutine 265 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
        sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:118 +0x1f4
panic({0x1b05880, 0xc0005e3020})
        runtime/panic.go:838 +0x207
sigs.k8s.io/security-profiles-operator/internal/pkg/manager/workloadannotator.(*PodReconciler).Reconcile(0xc000746300, {0xc000060c00?, 0xc000036030?}, {{{0xc000049b20?, 0x10?}, {0xc000049b00?, 0x40f3e7?}}})
        sigs.k8s.io/security-profiles-operator/internal/pkg/manager/workloadannotator/workloadannotator.go:122 +0xff2
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1f3be68?, {0x1f3bf10?, 0xc000036030?}, {{{0xc000049b20?, 0x1b3c160?}, {0xc000049b00?, 0x404ad4?}}})
        sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:121 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00013d9a0, {0x1f3be68, 0xc00002fec0}, {0x1a3c9c0?, 0xc000444020?})
        sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:320 +0x33c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00013d9a0, {0x1f3be68, 0xc00002fec0})
        sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:273 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:234 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:230 +0x325
```

#### What you expected to happen:

I expect the operator to function after installation, not immediately crash.

#### How to reproduce it (as minimally and precisely as possible):
```
$ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/security-profiles-operator/v0.4.3/deploy/operator.yaml
```

#### Anything else we need to know?:
Nextcloud has been rolled out with Helm (with MariaDB and Redis), with pretty basic settings. I initially thought that the zero in the name caused the issue, but Redis is also a statefulset and it doesn't complain about the zero in it's name.

Running on Ubuntu 22.04 with k3s.
```
NAME                    STATUS   ROLES                  AGE   VERSION        INTERNAL-IP      EXTERNAL-IP   OS-IMAGE           KERNEL-VERSION      CONTAINER-RUNTIME
MY_PRETTY_HOSTNAME   Ready    control-plane,master   61d   v1.23.6+k3s1   192.168.178.43   <none>        Ubuntu 22.04 LTS   5.15.0-37-generic   containerd://1.5.9-0ubuntu3
```

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6+k3s1", GitCommit:"418c3fa858b69b12b9cefbcff0526f666a6236b9", GitTreeState:"clean", BuildDate:"2022-04-28T22:16:18Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6+k3s1", GitCommit:"418c3fa858b69b12b9cefbcff0526f666a6236b9", GitTreeState:"clean", BuildDate:"2022-04-28T22:16:18Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
```

</details>


### Cloud provider

<details>
N.a.
</details>


### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
$ uname -a
Linux MY_PRETTY_HOSTNAME 5.15.0-35-generic #36-Ubuntu SMP Sat May 21 02:24:07 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

</details>


### Install tools

<details>

</details>


### Container runtime (CRI) and version (if applicable)

<details>
 containerd://1.5.9-0ubuntu3
</details>


### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>
Calico
</details>

If I move the seccomp profile to a folder:
```
seccompProfile:
    type: Localhost
    localhostProfile: nextcloud/redis-seccomp-profile.json
```

Nothing really changes. This time it's Redis, because I patched that one first.
```
I0610 17:50:30.331382       1 controller.go:117]  "msg"="Observed a panic in reconciler: runtime error: index out of range [2] with length 2" "controller"="pods" "controllerGroup"="" "controllerKind"="Pod" "name"="redis-master-0" "namespace"="nextcloud" "pod"={"name":"redis-master-0","namespace":"nextcloud"} "reconcileID"="7131aa4c-6d02-4241-a4b5-b57c6dda9716" 
panic: runtime error: index out of range [2] with length 2 [recovered]
        panic: runtime error: index out of range [2] with length 2

goroutine 298 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
        sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:118 +0x1f4
panic({0x1b05880, 0xc0006eedc8})
        runtime/panic.go:838 +0x207
sigs.k8s.io/security-profiles-operator/internal/pkg/manager/workloadannotator.(*PodReconciler).Reconcile(0xc000721a80, {0xc00037a800?, 0xc000741950?}, {{{0xc00062fbe0?, 0x10?}, {0xc00062fbc0?, 0x40f3e7?}}})
        sigs.k8s.io/security-profiles-operator/internal/pkg/manager/workloadannotator/workloadannotator.go:123 +0xfe5
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1f3be68?, {0x1f3bf10?, 0xc000741950?}, {{{0xc00062fbe0?, 0x1b3c160?}, {0xc00062fbc0?, 0x404ad4?}}})
        sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:121 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00030c500, {0x1f3be68, 0xc000721640}, {0x1a3c9c0?, 0xc0003fef40?})
        sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:320 +0x33c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00030c500, {0x1f3be68, 0xc000721640})
        sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:273 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:234 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:230 +0x325
```

```
# find /var/lib/kubelet/seccomp/
/var/lib/kubelet/seccomp/
/var/lib/kubelet/seccomp/transmission
/var/lib/kubelet/seccomp/transmission/transmission-seccomp-profile.json
/var/lib/kubelet/seccomp/nzbget
/var/lib/kubelet/seccomp/default
/var/lib/kubelet/seccomp/default/profile-complain-block-high-risk.json
/var/lib/kubelet/seccomp/default/profile-complain-unsafe.json
/var/lib/kubelet/seccomp/nextcloud
/var/lib/kubelet/seccomp/nextcloud/mariadb-seccomp-profile.json
/var/lib/kubelet/seccomp/nextcloud/nginx-seccomp-profile.json
/var/lib/kubelet/seccomp/nextcloud/nextcloud-seccomp-profile.json
/var/lib/kubelet/seccomp/nextcloud/redis-seccomp-profile.json
```

--- a/internal/pkg/manager/workloadannotator/workloadannotator.go
+++ b/internal/pkg/manager/workloadannotator/workloadannotator.go
@@ -43,6 +43,7 @@ const (
 	linkedPodsKey     = ".metadata.activeWorkloads"
 	StatusToProfLabel = "spo.x-k8s.io/profile-id"
 	reconcileTimeout  = 1 * time.Minute
+	pathParts         = 3
 )
 
 // NewController returns a new empty controller instance.
@@ -119,6 +120,9 @@ func (r *PodReconciler) Reconcile(_ context.Context, req reconcile.Request) (rec
 	// pod is being created or updated so ensure it is linked to a seccomp/selinux profile
 	for _, profileIndex := range getSeccompProfilesFromPod(pod) {
 		profileElements := strings.Split(profileIndex, "/")
+		if len(profileElements) != pathParts {
+			continue
+		}
 		profileNamespace := profileElements[1]
 		profileName := strings.TrimSuffix(profileElements[2], ".json")
 		seccompProfile := &seccompprofileapi.SeccompProfile{}
@@ -263,7 +267,11 @@ func getSeccompProfilesFromPod(pod *corev1.Pod) []string {
 	annotation, hasAnnotation := pod.GetAnnotations()[corev1.SeccompPodAnnotationKey]
 	if hasAnnotation && strings.HasPrefix(annotation, "localhost/") {
 		profileString := strings.TrimPrefix(annotation, "localhost/")
-		if !util.Contains(profiles, profileString) {
+		spCheck := &corev1.SeccompProfile{
+			Type:             "Localhost",
+			LocalhostProfile: &profileString,
+		}
+		if !util.Contains(profiles, profileString) && isOperatorSeccompProfile(spCheck) {
 			profiles = append(profiles, profileString)
 		}
 	}
@@ -311,7 +319,6 @@ func isOperatorSeccompProfile(sp *corev1.SeccompProfile) bool {
 	if !strings.HasSuffix(*sp.LocalhostProfile, ".json") {
 		return false
 	}
-	const pathParts = 3
 	return len(strings.Split(*sp.LocalhostProfile, "/")) == pathParts
 }
 

--- a/internal/pkg/manager/workloadannotator/workloadannotator_test.go
+++ b/internal/pkg/manager/workloadannotator/workloadannotator_test.go
@@ -110,6 +110,47 @@ func TestGetSeccompProfilesFromPod(t *testing.T) {
 			want: []string{profilePath},
 		},
 		{
+			name: "SeccompProfileRuntimeDefaultForPod",
+			pod: corev1.Pod{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{Name: "container1", Image: "testimage"}},
+					SecurityContext: &corev1.PodSecurityContext{
+						SeccompProfile: &corev1.SeccompProfile{
+							Type: "RuntimeDefault",
+						},
+					},
+				},
+			},
+			want: []string{},
+		},
+		{
+			name: "SeccompProfileLocalhostNoSlash",
+			pod: corev1.Pod{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{Name: "container1", Image: "testimage"}},
+					SecurityContext: &corev1.PodSecurityContext{
+						SeccompProfile: &corev1.SeccompProfile{
+							Type:             "Localhost",
+							LocalhostProfile: &[]string{"mariadb-seccomp-profile.json"}[0],
+						},
+					},
+				},
+			},
+			want: []string{},
+		},
+		{
+			name: "SeccompProfileInAnnotationNoSlash",
+			pod: corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{corev1.SeccompPodAnnotationKey: "localhost/mariadb-seccomp-profile.json"},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{Name: "container1", Image: "testimage"}},
+				},
+			},
+			want: []string{},
+		},
+		{
 			name: "SeccompProfileInPodAndContainerAndAnnotation",
 			pod: corev1.Pod{
 				ObjectMeta: metav1.ObjectMeta{

